velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -1002,7 +1002,13 @@ class ParquetRowReader::Impl {
       if (rowGroupInRange && !isExcluded && !isEmpty) {
         rowGroupIds_.push_back(i);
         firstRowOfRowGroup_.push_back(rowNumber);
+      } else if (i != 0) {
+        // Clear the metadata of row groups that are not read. This helps reduce
+        // the memory consumption. ColumnChunks consume the most memory.
+        // Skip the 0th RowGroup as it is used by estimatedRowSize().
+        rowGroups_[i].columns.clear();
```

Comment:
Surprise that metadata uses non-trivial amount of memory.  Is this a super wide table?

Reply:
If a parquet file has many rowgroups, we see this issue as well. 
When testing with SF10k and SF100k data, the unaccounted memory usage was significant.